homeassistant/components/google_generative_ai_conversation/entity.py

```diff
@@ -480,3 +489,67 @@ def create_generate_content_config(self) -> GenerateContentConfig:
                 ),
             ],
         )
+
+
+async def async_prepare_files_for_prompt(
+    hass: HomeAssistant, client: Client, files: list[Path]
+) -> list[File]:
+    """Append files to a prompt.
+
+    Caller needs to ensure that the files are allowed.
+    """
+    prompt_parts: list[File] = []
+
+    def append_files_to_prompt():
+        for filename in files:
+            if not filename.exists():
+                raise HomeAssistantError(f"`{filename}` does not exist")
+            mimetype = mimetypes.guess_type(filename)[0]
+            prompt_parts.append(
+                client.files.upload(
+                    file=filename,
+                    config={
+                        "mime_type": mimetype,
+                        "display_name": filename.name,
+                    },
+                )
+            )
```

Comment:
I think this no longer needs to append

```suggestion
    def upload_files() -> list[File]
        prompt_parts: list[File] = []
        for filename in files:
            if not filename.exists():
                raise HomeAssistantError(f"`{filename}` does not exist")
            mimetype = mimetypes.guess_type(filename)[0]
            prompt_parts.append(
                client.files.upload(
                    file=filename,
                    config={
                        "mime_type": mimetype,
                        "display_name": filename.name,
                    },
                )
            )
        return prompt_parts
```